### PR TITLE
Bump minimum Python version to 3.8

### PR DESCRIPTION
**Pull request by danielmitterdorfer**
With this commit we require at least Python 3.8 for Rally. This is done
as a preparation for Rally's new load generator which requires certain
Python 3.8 features (contextvars, madvise) and uses asyncio internally 
which has been improved in more recent versions of Python.

### CODE REVIEW

File: docs/install.rst [hunk 1]
```diff
@@ -13,35 +13,27 @@ Prerequisites
...
 
-**Debian / Ubuntu**
-
-::
-
-    sudo apt-get install gcc python3-pip python3.6-dev
+We recommend to use `pyenv <https://github.com/pyenv/pyenv>`_ to manage installation of Python. For details refer to their `installation instructions <https://github.com/pyenv/pyenv#installation>`_. Once ``pyenv`` is installed, install a compatible Python version::
 
+    # Install Python
+    pyenv install 3.8.0
```
Comment (dliappis) [inline docs/install.rst]: Should we mention specifically the system package prerequisites(for compiling python) as mandatory? These are listed nicely here: https://github.com/pyenv/pyenv/wiki/common-build-problems#prerequisites

Without the packages present, one can hit issues like:

```
$ pip3 install esrally
Collecting esrally
  Using cached https://files.pythonhosted.org/packages/eb/ca/371dcc49c7abcdb1bc8fec8aa76467658c891d935f0c2967137ff036a324/esrally-1.4.1-py3-none-any.whl
Collecting py-cpuinfo==3.2.0 (from esrally)
  Downloading https://files.pythonhosted.org/packages/48/1d/404dff45f8bdc63063ac0b822b557a5a5df54679de55261021efc882b8dd/py-cpuinfo-3.2.0.tar.gz (76kB)
     |████████████████████████████████| 81kB 4.7MB/s
    ERROR: Command errored out with exit status 1:
     command: /home/vagrant/.pyenv/versions/3.8.0/bin/python3.8 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-h35_rza2/py-cpuinfo/setup.py'"'"'; __file__='"'"'/tmp/pip-install-h35_rza2/py-cpuinfo/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base pip-egg-info
         cwd: /tmp/pip-install-h35_rza2/py-cpuinfo/
    Complete output (11 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/home/vagrant/.pyenv/versions/3.8.0/lib/python3.8/site-packages/setuptools/__init__.py", line 20, in <module>
        from setuptools.dist import Distribution, Feature
      File "/home/vagrant/.pyenv/versions/3.8.0/lib/python3.8/site-packages/setuptools/dist.py", line 35, in <module>
        from setuptools import windows_support
      File "/home/vagrant/.pyenv/versions/3.8.0/lib/python3.8/site-packages/setuptools/windows_support.py", line 2, in <module>
        import ctypes
      File "/home/vagrant/.pyenv/versions/3.8.0/lib/python3.8/ctypes/__init__.py", line 7, in <module>
        from _ctypes import Union, Structure, Array
    ModuleNotFoundError: No module named '_ctypes'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

which are a bit hard to diagnose.

Comment (danielmitterdorfer) [reply docs/install.rst]: I pondered doing this but this is also linked from Pyenv's installation instructions which is the reason I thought of not replicating it here. But you have a point...

Comment (danielmitterdorfer) [reply docs/install.rst]: addressed in https://github.com/elastic/rally/pull/934/commits/60193d8ba2e469466f1d8b73a542d04fa91d2971

Comment (dliappis) [reply docs/install.rst]: Thanks. I'd personally emphasize with bold as below, but it's a personal preference (to ensure maximum visibility for users)

```
and **ensure that all of** `pyenv's prerequisites <https://github.com/pyenv/pyenv/wiki/common-build-problems#prerequisites>`_ are installed.
```

Comment (danielmitterdorfer) [reply docs/install.rst]: Addressed in https://github.com/elastic/rally/pull/934/commits/726f89062ca7cb2b5fdfae2e6a3e48011fdb0846.

**Pull request merged**